main.js

```diff
@@ -78,6 +78,14 @@ function emptyConnectionForm() {
 }
 
 function connect(id) {
+  if(!id) {
```

Comment:
We can get the formdata outside of the condition and then have one combined condition that checks 
```
if (!id && (!formData.connectionOptions.url || !formData.connectionOptions.port)) {
   showConnectionError('some text');
   return;
}
```

Reply:
In that case we would always collect the form data (which includes slow jQuery selectors) even if we don't need them.